src/main/java/org/broadinstitute/hellbender/utils/read/ReadCoordinateComparator.java

```diff
@@ -0,0 +1,71 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.SAMFileHeader;
+
+import java.util.Comparator;
+
+/**
+ * Comparator for sorting Reads by coordinate. Note that a header is required in
+ * order to meaningfully compare contigs.
+ *
+ * Uses the various other fields in a Read to break ties for Reads that share
+ * the same location.
+ *
+ * Based loosely on the SAMRecordCoordinateComparator from htsjdk.
+ */
+public class ReadCoordinateComparator implements Comparator<GATKRead> {
```

Comment:
can you simplify this code by using methods on the Comparator interface, such as `comparingInt`, `thenComparing` etc?